File: 백준/Silver/12101. 1， 2， 3 더하기 2/1， 2， 3 더하기 2.py
import sys

input = sys.stdin.readline

n,k = map(int,input().split())
result = []

def dfs(total,array=[]):
  if(total > n):return
  if(total ==n):
    result.append(array[:])
    return
  for i in range(1,4):
    array.append(i)
    dfs(total+i,array)
    array.pop()

dfs(0,)
def makeString(array):
  return '+'.join(list(map(str,array)))
print(makeString(result[k-1]) if len(result) >=k else -1)

File: 백준/Silver/12101. 1， 2， 3 더하기 2/1， 2， 3 더하기 2.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
const [n, k] = input[0].split(" ").map(Number)

const num = [1, 2, 3]
const solution = (target) => {
  const numbers = []
  const getNum = (target, array) => {
    const total = array.reduce((a, b) => a + b, 0)
    if (total > target) {
      return
    }
    if (total == target) {
      numbers.push(array.join(""))
    }
    for (let i = 0; i < 3; i++) {
      array.push(num[i])
      getNum(target, array)
      array.pop()
    }
  }
  getNum(target, [])
  numbers.sort((a, b) => (a < b ? -1 : 1))
  return numbers
}

const answer = solution(n).map((val)=>val.split('').join('+'))[k-1] ?? -1
console.log(answer)
